src/Lang/Surface.ml

```diff
@@ -249,10 +262,19 @@ and def_data =
   | DDataRec of data_def list
     (** Definition of mutually recursive ADTs *)
 
+  | DModule of module_name * def list
+    (** Definition of a module *)
+
+  | DOpen of module_name path
+    (** Opening a module *)
+
   | DReplExpr of expr
     (** Print type, evaluate, and print the expression, provided by a user in
       REPL. *)
 
+  | DPub of def
```

Comment:
I understand that this approach is somehow clean (we didn't have to touch all the existing constructors), but I'm not sure if it would scale to e.g., `abstract` visibility modifier of datatypes, proposed in this review.

Reply:
See the comments to `Pattern.mli`.